src/api/spec/requests/kerberos_login_spec.rb

```diff
@@ -1,82 +1,64 @@
         before do
-          user.update(state: 'unconfirmed')
+          get "/source.xml"
         end
 
-        it 'does not authenticate the user' do
-          get "/source.xml", headers: { 'X-HTTP_AUTHORIZATION' => "Negotiate #{Base64.strict_encode64('fake_ticket')}" }
-          expect(response).to have_http_status(:forbidden)
-          expect(response.body).to match('User is registered but not yet approved. Your account is a registered account, ' +
-            'but it is not yet approved for the OBS by admin.')
-        end
+        it { expect(response).to have_http_status(:unauthorized) }
+        it { expect(response.header["WWW-Authenticate"]).to eq('Negotiate') }
       end
 
-      context 'but for a diferent realm' do
+      context "authorization header contains 'Negotiate' with a ticket" do
+        let(:gssapi_mock) { double(:gssapi) }
+
         before do
+          allow(gssapi_mock).to receive(:acquire_credentials)
+          allow(gssapi_mock).to receive(:accept_context).
+            with('ticket').and_return(true)
           allow(gssapi_mock).to receive(:display_name).
-            and_return("#{user.login}@my_other_realm.com")
+            and_return("#{user.login}@test_realm.com")
 
           allow(GSSAPI::Simple).to receive(:new).with(
             'obs.test.com', 'HTTP', '/etc/krb5.keytab'
           ).and_return(gssapi_mock)
         end
 
-        it 'does not authenticate the user' do
-          get "/source.xml", headers: { 'X-HTTP_AUTHORIZATION' => "Negotiate #{Base64.strict_encode64('fake_ticket')}" }
-          expect(response).to have_http_status(:unauthorized)
-          expect(response.body).to match('User authenticated in wrong Kerberos realm.')
+        it 'authenticates the user' do
+          get "/source.xml", headers: { 'X-HTTP_AUTHORIZATION' => "Negotiate #{Base64.strict_encode64('ticket')}" }
+          expect(response).to have_http_status(:success)
         end
       end
-    end
 
-    context 'with invalid ticket' do
-      before do
-        allow(gssapi_mock).to receive(:acquire_credentials).and_raise(GSSAPI::GssApiError)
-        allow(GSSAPI::Simple).to receive(:new).with(
-          'obs.test.com', 'HTTP', '/etc/krb5.keytab'
-        ).and_return(gssapi_mock)
-      end
+      context 'authenticator raises an error while fetching user' do
+        let(:auth_header) { "Negotiate #{Base64.strict_encode64('ticket')}" }
 
-      it 'does not authenticate the user' do
-        get "/source.xml", headers: { 'X-HTTP_AUTHORIZATION' => "Negotiate #{Base64.strict_encode64('fake_ticket')}" }
-        expect(response).to have_http_status(:unauthorized)
-        expect(response.body).to match('Received a GSSAPI exception')
+        before do
+          allow_any_instance_of(Authenticator).to receive(:extract_krb_user).
+            with(auth_header.to_s.split).
+            and_raise(Authenticator::AuthenticationRequiredError, 'something happened')
```

Comment:
Well, the authenticator class get's tested intensively (at least the kerberos related parts...). So this should be fine.